Model/CheckoutSessionManagement.php

```diff
@@ -720,83 +721,99 @@ private function cancelOrder($order, $quote)
      */
     public function completeCheckoutSession($amazonSessionId, $cartId = null)
```

Comment:
Could some room be given to the different logic in this method? Some areas just run together and it si a bit hard to digest.